feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/SessionDetailFragment.kt

```diff
@@ -137,194 +138,63 @@ class SessionDetailFragment : DaggerFragment() {
-                clickListener()
-            }
-
-            override fun updateDrawState(ds: TextPaint) {
-                // nothing
-            }
-        }, builderAction)
-    }
-
-    private fun ViewGroup.bindSpeaker(session: Session) {
-        removeAllViews()
-        (session as? SpeechSession)?.speakers.orEmpty().indices.forEach { index ->
-            val speaker: Speaker =
-                (session as? SpeechSession)?.speakers?.getOrNull(index) ?: return@forEach
-            val speakerView = layoutInflater.inflate(
-                R.layout.layout_speaker_session_detail, this, false
-            ) as ViewGroup
-            val speakerNameView = speakerView.findViewById<TextView>(R.id.speaker)
-            val speakerImageView = speakerView.findViewById<ImageView>(R.id.speaker_image)
-            speakerImageView.transitionName = "${speaker.id}-${TRANSITION_NAME_SUFFIX}"
-            speakerView.setOnClickListener {
-                val extras = FragmentNavigatorExtras(
-                    speakerImageView to speakerImageView.transitionName
+        if (binding.sessionDetailRecycler.adapter == null) {
+            val adapter = GroupAdapter<ViewHolder<*>>()
+            binding.sessionDetailRecycler.adapter = adapter
+            binding.sessionDetailRecycler.layoutManager = LinearLayoutManager(context)
+            context?.let {
+                binding.sessionDetailRecycler.addItemDecoration(
+                    SessionDetailItemDecoration(
+                        adapter,
+                        it
+                    )
                 )
-                findNavController()
-                    .navigate(actionSessionToSpeaker(speaker.id, TRANSITION_NAME_SUFFIX), extras)
             }
-            bindSpeakerData(speaker, speakerNameView, speakerImageView)
-            addView(speakerView)
-        }
-    }
-
-    private fun bindSpeakerData(
-        speaker: Speaker,
-        speakerNameView: TextView,
-        speakerImageView: ImageView
-    ) {
-        speakerNameView.text = speaker.name
-//        setHighlightText(textView, query)
-        val imageUrl = speaker.imageUrl
-        val context = speakerNameView.context
-        val placeHolder = run {
-            VectorDrawableCompat.create(
-                context.resources,
-                R.drawable.ic_person_outline_black_32dp,
-                null
-            )?.apply {
-                setTint(
-                    context.getThemeColor(R.attr.colorOnBackground)
-                )
-            }
-        }?.also {
-            speakerImageView.setImageDrawable(it)
-        }
-
-        Coil.load(context, imageUrl) {
-            crossfade(true)
-            placeholder(placeHolder)
-            transformations(CircleCropTransformation())
-            lifecycle(viewLifecycleOwner)
-            target {
-                speakerImageView.setImageDrawable(it)
+            adapter.add(sessionDetailTitleItemFactory.create(session))
```

Comment:
In order to enable DiffUtil of RecyclerView, it is necessary to make a list as follows and update it.

```
val items = mutableListOf<Group>()
items += sessionDetailDescriptionItemFactory.create(session)
...
adapter.update(items)
```